src/services/services.ts

```diff
@@ -6804,6 +6795,74 @@ namespace ts {
             return [];
         }
 
+        /**
+         * Checks if position points to a valid position to add JSDoc comments, and if so,
+         * returns the appropriate template. Otherwise returns an empty string.
+         * Valid positions are
+         * * outside of comments, statements, and expressions, and
+         * * preceding a function declaration.
+         *
+         * Hosts should ideally check that:
+         * * The line is all whitespace up to 'position' before performing the insertion.
+         * * If the keystroke sequence "/\*\*" induced the call, we also check that the next

```

Comment:
The backslashes escape the asterisk when it is rendered by hovering over the definition or a call of `getDocCommentTemplateAtPosition()`.